### PR TITLE
Expand native number compaction edge case tests

### PR DESCRIPTION
**Pull request by hmottestad**
## Summary
- document a dozen native number compaction edge cases directly in the regression test class
- add dedicated coverage for set, index, preserve, multi-value, and competing term scenarios using native numbers

## Testing
- mvn -Dtest=UriCompactionNativeTypesTest test

------
https://chatgpt.com/codex/tasks/task_e_68eab7e312a8832e91657b62c14069bd

### CODE REVIEW

File: src/main/java/no/hasmac/jsonld/compaction/UriCompaction.java [hunk 1]
```diff
@@ -339,6 +339,16 @@ private String compact_4(String variable) throws JsonLdError {
                 // 4.7.8.
             } else {
                 typeLanguageValue = commonLanguage;
+
```
Comment (nk2IsHere) [inline src/main/java/no/hasmac/jsonld/compaction/UriCompaction.java]: I've been looking into the fix here, and into the spec (https://www.w3.org/TR/json-ld-api/#compaction-algorithm)
This fix does fix the original issue on the surface, but unfortunately it si non-conformant to the specification of the algorithm.
Our guess that it will, in future, make the project less and less conformant with the original jsonld spec -> less maintainable.

File: src/main/java/no/hasmac/jsonld/compaction/UriCompaction.java [hunk 2]
```diff
@@ -339,6 +339,16 @@ private String compact_4(String variable) throws JsonLdError {
...
+                if (Keywords.NULL.equals(typeLanguageValue)
+                        && listContainsNativeNumberValues(list)) {
```
Comment (nk2IsHere) [inline src/main/java/no/hasmac/jsonld/compaction/UriCompaction.java]: There is another question from us, whether to the algorithm of the fix, or to the naming of the methods added:
- It says "listContainsNativeNumberValues" Numbers? Does it only apply fix if it detects a compaction pattern on the typed numbers (floats, doubles, other `xsd:....`)?
- What about booleans, language strings, other types, maybe even outside of scope of xsd namespace for literals?

I do believe it should be supported in generic fashion. Seems original code should be capable doing it out of the box.

File: src/main/java/no/hasmac/jsonld/compaction/UriCompaction.java (hunk 1 above)
Comment (nk2IsHere) [reply src/main/java/no/hasmac/jsonld/compaction/UriCompaction.java]: We have noticed ourselves, that the problem does not come from how compaction algorithm is implemented, but from earlier algorithm of inverse context creation:

Please look here (its an inverse context generated on our jsonld):

This is a term with type
image


This is a term without type
image

`@language` tag is missing.

Can we please explore this direction? Seems fixes for compaction algorithm are either to spec itself, or to deeper parts that cause it

File: src/main/java/no/hasmac/jsonld/compaction/UriCompaction.java [hunk 3]
```diff
@@ -556,4 +582,65 @@ private String compact_4(String variable) throws JsonLdError {
...
+                inferredType = typeMapping;
+            } else if (!inferredType.equals(typeMapping)) {
+                return null;
+            }
+        }
+
+        return inferredType;
+    }
+
+    private boolean listContainsNativeNumberValues(final JsonArray list) {
```
Comment (nk2IsHere) [inline src/main/java/no/hasmac/jsonld/compaction/UriCompaction.java]: It seems this method has nothing to do with numbers. Codex picked up numbers topic but made it still generic?

File: src/test/java/no/hasmac/jsonld/compaction/UriCompactionNativeTypesTest.java [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+     * 3. Native number compaction respects default language and direction settings.
+     * 4. Native number with an @index container selects the correct term alias.
+     * 5. Native number with an @set container is compacted using the typed term.
+     * 6. Native number with combined @set and @index containers keeps the typed alias.
+     * 7. Native number lists preserved via @preserve still resolve to the typed alias.
+     * 8. Native number lists with multiple entries consistently use the typed alias.
+     * 9. Native number that is preserved via @preserve unwraps and keeps the typed term.
+     * 10. Native number on a value object that also carries an @index attribute retains the term.
+     * 11. Native number when multiple terms share the IRI prefers the typed definition.
+     * 12. Native number prefers the typed term even when CURIE compaction is possible.
```
Comment (nk2IsHere) [inline src/test/java/no/hasmac/jsonld/compaction/UriCompactionNativeTypesTest.java]: What is CURIE compaction and how does it relate to normal compaction. Googling shows no such compaction exists. Compaction under test case is just a normal compaction. Can we please remove it?

File: src/test/java/no/hasmac/jsonld/compaction/UriCompactionNativeTypesTest.java [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class UriCompactionNativeTypesTest {
+
+    /*
+     * Edge cases covered:
+     * 1. Direct native number with explicit type mapping prefers the scoped term.
+     * 2. Native number inside an @list container still resolves to the typed term.
```
Comment (nk2IsHere) [inline src/test/java/no/hasmac/jsonld/compaction/UriCompactionNativeTypesTest.java]: There are many tests here that essentially attach some attribute to the original term mapping and see if it maps properly. In the original algorithm, to be frank, we learned that none of such attributes (like `@preserve`, `@index`) matter, except `@language`. Are these tests necessary? Do they produce proper edge cases that need to be handled?

Why are there no tests for other types of terms, only numbers? Could we please explore this option?

(PS: We have the same problem with `xsd: boolean`, so... seems to be a general type issue, not just numbers)

File: src/main/java/no/hasmac/jsonld/compaction/UriCompaction.java (hunk 1 above)
Comment (nk2IsHere) [reply src/main/java/no/hasmac/jsonld/compaction/UriCompaction.java]: I will try to cook up my own fix, in inverse context realm of things. it should cover types automatically, since compact4 is by spec
